wgpu-core/src/device/resource.rs

```diff
@@ -718,7 +718,18 @@ impl<A: HalApi> Device<A> {
             if desc.format == *format {
                 continue;
             }
-            if desc.format.remove_srgb_suffix() != format.remove_srgb_suffix() {
+
+            if !matches!(
+                (desc.format, *format),
+                (
+                    TextureFormat::NV12,
+                    TextureFormat::R8Unorm
+                        | TextureFormat::R8Uint
+                        | TextureFormat::Rg8Unorm
+                        | TextureFormat::Rg8Uint,
+                )
+            ) && desc.format.remove_srgb_suffix() != format.remove_srgb_suffix()
```

Comment:
Could we move this logic into a function on format that determines if a given view format is compatible with the texture format.

Reply:
Done.